src/main/java/org/gaul/s3proxy/Main.java

```diff
@@ -117,14 +121,44 @@ public static void main(String[] args) throws Exception {
 
             String s3ProxyAuthorizationString = properties.getProperty(
                     S3ProxyConstants.PROPERTY_AUTHORIZATION);
+            ImmutableList.Builder<String> locatorBuckets =
+                    new ImmutableList.Builder<>();
+            for (String key : properties.stringPropertyNames()) {
+                if (key.startsWith(S3ProxyConstants.PROPERTY_BUCKET_LOCATOR)) {
+                    locatorBuckets.add(properties.getProperty(key));
+                }
+            }
+            ImmutableList<String> buckets = locatorBuckets.build();
+            String localIdentity = null;
+            String localCredential = null;
             if (AuthenticationType.fromString(s3ProxyAuthorizationString) !=
                     AuthenticationType.NONE) {
-                String localIdentity = properties.getProperty(
+                localIdentity = properties.getProperty(
                         S3ProxyConstants.PROPERTY_IDENTITY);
-                String localCredential = properties.getProperty(
+                localCredential = properties.getProperty(
                         S3ProxyConstants.PROPERTY_CREDENTIAL);
-                locators.put(localIdentity, Maps.immutableEntry(
-                        localCredential, blobStore));
+                Map.Entry<String, String> key = Maps.immutableEntry(
+                        localIdentity, null);
+                if (!parsedLocations.contains(key)) {
```

Comment:
You can use the return value from `parsedLocations.add`.

Reply:
Done